src/main/java/com/google/sps/data/GivrUser.java

```diff
@@ -104,18 +104,19 @@ public static void updateUserInDatastore(String identifyingProperty, String iden
     datastore.put(entity);
   }
 
-  public static GivrUser getUserById(String userId) {
-    Entity entity = getUserFromDatastoreWithProperty("userId", userId);
+  public static GivrUser getUserByIdOrEmail(String userId, String userEmail) {
+    Entity entityRetrievedWithId = getUserFromDatastoreWithProperty("userId", userId);
+    Entity entityRetrievedWithEmail = getUserFromDatastoreWithProperty("userEmail", userEmail);
```

Comment:
So before this change there was:
 getUserById
 getUserByEmail
 getCurrentLoggedInUser

After this change there is a:
 getUserByIdOrEmail
 getUserByEmail
 getCurrentLoggedInUser

The "before" seems like a better API to me than the after, but I understand why you want to merge these cases. What if you kept the before API, but implemented both of those functions via a new private helper: getUserByIdOrEmail

Reply:
This makes sense! I'll add it back.